lib/31XUtils/schemaUtils31X.js

```diff
@@ -154,9 +154,11 @@ module.exports = {
     const resolvedSchema = _.cloneDeep(refSchema),
       outerKeys = Object.keys(outerProps);
     outerKeys.forEach((key) => {
-      resolvedSchema[key] = (resolvedSchema[key] && Array.isArray(resolvedSchema[key])) ?
-        [...new Set([...resolvedSchema[key], ...outerProps[key]])] :
-        outerProps[key];
+      if (_.isObject(resolvedSchema) && _.isObject(outerProps)) {
```

Comment:
This check for outerProps(`_.isObject(outerProps)`) can be moved outside of the loop to prevent checking on every iteration.